app/src/main/java/com/example/wearit/ui/WardrobeScreen.kt

```diff
@@ -1,43 +1,120 @@
+//import androidx.compose.foundation.layout.ColumnScopeInstance.align
+//import androidx.compose.foundation.layout.ColumnScopeInstance.align
+//import androidx.compose.foundation.layout.ColumnScopeInstance.align
+//import androidx.compose.foundation.layout.ColumnScopeInstance.align
+//import androidx.compose.foundation.layout.ColumnScopeInstance.align
+//import androidx.compose.foundation.layout.ColumnScopeInstance.align
+//import androidx.compose.foundation.layout.ColumnScopeInstance.align
+import androidx.compose.foundation.lazy.*
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
 import androidx.compose.material.*
 import androidx.compose.material.icons.Icons
-import androidx.compose.material.icons.filled.Favorite
-import androidx.compose.material.icons.filled.PlayArrow
+import androidx.compose.material.icons.filled.Menu
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.shadow
 import androidx.compose.ui.graphics.Color
-import androidx.compose.ui.text.font.Font
-import androidx.compose.ui.text.font.FontFamily
-import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.modifier.modifierLocalConsumer
+import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.unit.dp
-import androidx.compose.ui.unit.sp
-import com.example.wearit.R
 import com.example.wearit.model.Category
-import com.example.wearit.ui.theme.Purple500
+import com.example.wearit.model.Item
 import java.util.*
-
+import kotlin.collections.ArrayList
 
 @Composable
 fun WardrobeScreen(
     currentCategory: Category,
     goToPickerScreen: () -> Unit,
-    onCategoryChange: (category:Category) -> Unit,
+    onCategoryChange: (category: Category) -> Unit,
+    itemsOfCurrentCategory: List<Item>?,
+
+    ) {
 
-) {
     val listOfCategories = (Category.values())
 
+    val array = ArrayList<Item>()
+    if (itemsOfCurrentCategory != null) {
+        for (item in itemsOfCurrentCategory) {
+            array.add(item)
+        }
+    }
+
+    Scaffold(
+        content = { innerPadding ->
+            //giving padding to whole content (so it doesnt overlap with bottomBar
+            WardrobePageContent(
+                innerPadding = innerPadding,
+                onCategoryChange = onCategoryChange,
+                listOfCategories=listOfCategories,
+                itemsOfCurrentCategory = array
+                )
+        },
+            bottomBar = {
+                BottomBarSpace()
+            }
+    )
+}
 
-    Column {
-        Text(text = "Current category is " + currentCategory.name)
+@Composable
+fun WardrobePageContent(
+    innerPadding: PaddingValues,
+    onCategoryChange: (category: Category) -> Unit,
+    listOfCategories: Array<Category>,
+    itemsOfCurrentCategory: ArrayList<Item>
```

Comment:
Czy jest jakiś powód żeby używać Array<> i ArrayList<>? Wg mnie tu powinno być po prostu List<>. Być może to rozwiąże ten problem z przepisywaniem do nowej listy. A jeśli to by nie pomogło to spróbuj jeszcze dopisać !! w przekazywaniu argumentu jak wywołujesz komponent

Reply:
zmieniłem na list, jednak teraz pojawia sie problem - aplikacja się crashuje gdy próbuję wylistować itemki z kategorii która tych itemków nie posiada ( np shorts ) i nie wiem czym moze byc to spowodowane

Reply:
dobra, udało się go z kamilem naprawić, trzeba było dorobić kategorie w pliku fakeitemsdata, bo jak próbowaliśmy pobrać liste itemków z kategorii która nie istniała w fakeitemsdata to crashowalo